drv/vsc7448/src/dev.rs

```diff
@@ -0,0 +1,206 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use crate::{
+    port::{port10g_flush, port1g_flush},
+    spi::Vsc7448Spi,
+    VscError,
+};
+use vsc7448_pac::Vsc7448;
+
+/// DEV1G and DEV2G5 share the same register layout, so we can write functions
+/// that use either one.
+#[derive(Copy, Clone)]
+pub enum DevGeneric {
+    Dev1g(u32),
```

Comment:
It looks like the fields here have a range of 0..23 and 0..28, respectively; you could use `u8` here and make them cheaper to store. I don't know if you store them or anything though.

Reply:
(It's worth noting that all of the arithmetic you're doing on dev numbers below implies overflow checks -- storing a u8 and converting to a u32 before arithmetic _should_ make it clear to the compiler that overflow can't happen. This may or may not result in a text size decrease, depending on whether the compiler had already inlined enough things to see this on its own.)

Reply:
I updated the PAC to use the minimum-sized integers for indexes of things like `DEV1G`, and updated all of this code as well!